proxy/ProxyTransaction.h

```diff
@@ -85,6 +88,10 @@ class ProxyTransaction : public VConnection
   // Returns true if there is a request body for this request
   virtual bool has_request_body(int64_t content_length, bool is_chunked_set) const;
 
+  sockaddr const *get_remote_addr() const;
+
+  virtual HTTPVersion get_version(HTTPHdr &hdr) const;
```

Comment:
We don't need this at the moment.

Since`ProxySession::get_version` receives `HTTPHdr`, a proxy session can always return correct version. I wonder if there will be a session that has multiple versions of transactions. I'm not sure if we are allowed to use multiple versions on a traditional HTTP connection like below:

```
[Connect]
GET /a HTTP/1.0

GET /b HTTP/1.1
Host: example.com

[Close]
```

Reply:
I am not sure CONNECT makes my head hurt.  I would think that only one nested method would be interpreted over the CONNECT method.  And ATS would only interpret it if the CONNECT was made over a non-TLS connection.  Otherwise ATS treats the body as a blind tunnel.

Will have to read the RFC tea leaves on this.